### ISSUE TITLE
URL for Laravel Assets

### ISSUE THREAD

**Issue by buddhaCode** [brefphp/bref#1046]
Hi,
I am just trying to get my Laravel App running on AWS Lambda with Bref. Thanks for this great piece of software and documentation.

I think there are some inaccuracies with the assets and the Mix assets of Laravel. Please have a look at the documentation under Laravel > Assets (https://bref.sh/docs/frameworks/laravel.html#assets).

You state, that one should set the environment variable `MIX_ASSET_URL` to the S3 bucket URL and use the `asset()` function to include the assets and generate a URL for it. I think that's just one part of the story. You have to differentiate between assets generated with Mix/Webpack (like CSS or JS files) and other assets in the public folder (like images, documents, etc.). The Mix asset files a typically referenced in the templates with the `mix()` function. And that's where the `MIX_ASSET_URL` env var will be applied.
If you use the `asset()` function, one have to set the `ASSET_URL` env var to the S3 bucket URL, as well. Otherwise, the generated URL will not point to the S3 bucket, but to the Lambda URL.

As I follow the documentation as described, I get dead URLs.

In short:
`MIX_ASSET_URL` => `mix()`
 `ASSET_URL` => `asset()`

Can someone have a look at it and confirm my point of view?

**Comment by mnapoli** [comment]
Ohh interesting. @deleugpn any experience with that to confirm?

Thank you for reporting that!

**Comment by buddhaCode** [comment]
Let me give you some code snippets.

The `asset()` function is a shortcut for the `url` facade (`\Illuminate\Routing\UrlGenerator::asset()`). In this method an [asset root](https://github.com/laravel/framework/blob/8.x/src/Illuminate/Routing/UrlGenerator.php#L242) is taking into account. The facade is initialized in the [`RouteServiceProvider`](https://github.com/laravel/framework/blob/8.x/src/Illuminate/Routing/RoutingServiceProvider.php#L65). There you can see, that the asset root property is set to the `app.asset_url` config value. Which can be set this with the `ASSET_URL` env var.

The `mix()` function is an alias for `\Illuminate\Foundation\Mix::__invoke`. At the [end of the method](https://github.com/laravel/framework/blob/8.x/src/Illuminate/Foundation/Mix.php#L72) the `app.mix_url` config value is taking into account to generate the url. Which is [populated in the Bref Laravel bridge](https://github.com/brefphp/laravel-bridge/blob/master/src/BrefServiceProvider.php#L49) with the `MIX_ASSET_URL` env var.

I would suggest to set the `app.asset_url` config value to the `MIX_ASSET_URL` env var in the bridge as well.

**Comment by buddhaCode** [comment]
BTW, this[ Amazon blog post](https://aws.amazon.com/de/blogs/compute/the-serverless-lamp-stack-part-4-building-a-serverless-laravel-application/) under section "Referencing static assets in view templates" states the same issue with the `ASSET_URL` env var.

**Comment by mnapoli** [comment]
👍  thanks for the details!

A pull request would be most welcome :)

**Comment by t-richard** [comment]
This part of the docs has been rewritten recently to use Lift server-side construct instead of S3 directly.

Is all of this still relevant then?

See https://bref.sh/docs/frameworks/laravel.html#assets

**Comment by mnapoli** [comment]
😮  but of course yeah! Maybe the problem doesn't exist anymore!

**Comment by buddhaCode** [comment]
Oh yes, that's right. If you host your Lambda under a custom domain, the problem doesn't exist anymore.

I have to admit, that I do not host the static files under the same domain. I am just using the generic S3 URL and do not use the CloudFront distribution. Maybe this is a totally different case and not the standard case anymore.

**Comment by shabanbx** [comment]
i am having this same issue 
i used Lift for the assets

when in route53 i route my domain to the cloudfront the assets starts working on domain.com/css/app.css
but in laravel the asset url is still default-api-gateway-url.com/css/app.css on which the asset does not open


so when i replace the CloudDistribution A record with a custom domain api gateway 
in laravel the url of the assets are displayed correctly domain.com/css/app.css

but the assets won't open then

**Comment by t-richard** [comment]
This seems like your laravel app is not trusting the x-forwarded-host header.

You may look at this part of the Laravel docs to configure this https://laravel.com/docs/8.x/requests#configuring-trusted-proxies

You should probably something like this to the middleware

```
protected $headers = Request::HEADER_X_FORWARDED_FOR | Request::HEADER_X_FORWARDED_HOST | Request::HEADER_X_FORWARDED_PORT | Request::HEADER_X_FORWARDED_PROTO;
```

**Comment by shabanbx** [comment]
@t-richard 
thankyou so much just had to do 
 protected $proxies = '*'; 
in the TrustProxies and it worked

i think this should be in the docs though.

**Comment by t-richard** [comment]
@shabanbx feel free to send a PR 😉

**Comment by mnapoli** [comment]
I'll keep this one open as we still want to fix this (unless I missed something).

**Comment by mnapoli** [comment]
This should be fixed with v2!
